packages/react-aria-components/src/Menu.tsx

```diff
@@ -253,30 +246,48 @@ function MenuInner<T extends object>({props, collection, menuRef: ref}: MenuInne
             new KeyboardEvent(e.nativeEvent.type, e.nativeEvent)
           );
         }
+
         focusedId = getItemId(state, state.selectionManager.focusedKey);
         return focusedId;
       });
     }
   }, [register, state, menuId, ref]);
 
+  // Update the focused key to be the first item in the menu only if the input value changes (aka match spotlight/other implementations).
+  let updateFocusedKey = useEffectEvent(() => {
+    // TODO: the below is pretty much what the listkeyboard delegate would do when finding the first key
+    state.selectionManager.setFocused(true);
+    let focusedNode = state.collection.getItem(state.selectionManager.focusedKey);
+    if (focusedNode == null || focusedNode.prevKey != null) {
+      let key = state.collection.getFirstKey();
+      while (key != null) {
+        let item = state.collection.getItem(key);
+        if (item?.type === 'item' && !state.selectionManager.isDisabled(key)) {
+          break;
+        }
+        key = state.collection.getKeyAfter(key);
+      }
+
+      state.selectionManager.setFocusedKey(key);
+      setFocusedNodeId && setFocusedNodeId(key == null ? null : getItemId(state, key));
+    }
+  });
+
   useEffect(() => {
-    // TODO: retested in NVDA. It seems like NVDA properly announces what new letter you are typing even if we maintain virtual focus on
+    // TODO: retested in NVDA. It seems like NVDA properly announces what new letter you are typing/deleting even if we maintain virtual focus on
     // a item in the list. However, it won't announce the letter your cursor is now on if you don't clear the virtual focus when using left/right
-    // arrows.
-    // Clear the focused key if the inputValue changed for NVDA
-    // Also feels a bit weird that we need to make sure that the focusedId AND the selection manager here need to both be cleared of the focused
-    // item, would be nice if it was all centralized. Maybe a reason to go back to having the autocomplete hooks create and manage
-    // the collection/selection manager but then it might cause issues when we need to wrap a Table which won't use BaseCollection but rather has
-    // its own collection
+    // arrows because the NVDA blue focus indicator moves into the menu and is only restored back to the input when typing. In Voiceover it works just fine though,
+    // mobile screen readers don't suffer from this when using virtual keyboards either
+    // When adding the "auto focus the first item if no items are focused" behavior, the announcement of the letter typed gets cut off a bit in NVDA.
+    // Ariakit seems to announce a bit better, doesn't interrupt.
+    // Also it feels like we are pulling in more and more stuff/updating focused key in more places, might be nice if this was more centralized
```

Comment:
the announcement behavior in NVDA is actually bit more nuanced that this depending if the newly focused item is the same as before, will need more testing